### PR TITLE
[LeagueEvent] Remove unnecessary check instanceof StringType on DispatchStringToObjectRector

### PR DESCRIPTION
**Pull request by samsonasik**
Ref https://github.com/rectorphp/rector/pull/6075#discussion_r611030217 /cc @lulco

### CODE REVIEW

File: rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php [hunk 1]
```diff
@@ -91,13 +91,10 @@ private function shouldSkip(MethodCall $methodCall): bool
...
         }
 
-        if ($this->nodeTypeResolver->isObjectTypes($methodCall->var, [
+        return ! $this->nodeTypeResolver->isObjectTypes($methodCall->var, [
             new ObjectType('League\Event\EventDispatcher'),
             new ObjectType('League\Event\Emitter'),
-        ])) {
-            return false;
-        }
-        return $this->getStaticType($methodCall->args[0]->value) instanceof StringType;
```
Comment (lulco) [inline rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php:100]: I am not sure about this. If the value is variable, you have to pass it somehow to anonymous class or not?

Comment (samsonasik) [reply rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php:100]: it needs more fixture to proof that, it was always return true, which the check can be removed. see https://github.com/rectorphp/rector/blob/4b71eaa53d16c9c8cf4a6a2eb1fe4bf6d8006c58/rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php#L96-L100

Comment (lulco) [reply rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php:100]: Ok I will create some fixtures with different approaches

Comment (lulco) [reply rules/LeagueEvent/Rector/MethodCall/DispatchStringToObjectRector.php:100]: @samsonasik Failing test added https://github.com/rectorphp/rector/pull/6120
I'll try to add fix

**Pull request merged**